steveify1/multi-tenant-blog-api
# #7 feature:app permissions

#### What does this PR do?
- implement app permission management
#### Description of Task to be completed?
- implement the `AppPermissionService` and controller
- implement the `POST /app-permissions` endpoint
- implement the `GET /app-permissions` endpoint
- implement the `GET /app-permissions/:id` endpoint
- implement the `PATCH /app-permissions/:id` endpoint
- implement a `verifyRowExists` method to the `CreateAndRead` service class
- document the endpoints with postman
- write tests for the endpoints
- write a unit test for the `AppPermissionService` class
#### How should this be manually tested?
* clone the repository
* cd into the project directory
* run `npm install`
* run `npm test` to run all the tests (not required)
* run `npm run dev` to start the development server
* send a `POST` request to the `/app-permissions` route to define a new permission. The request body should be in the following format
```
{
 "title": <string>,
 "max_authors": <integer>,
 ...
}
```
Please see the documentation for a full (growing) list of possible permission definitions

#### Any background context you want to provide?
* The base URL of the API is `http://localhost:5000/api/v1`
* Two Postgresql databases, `mblog` and `mblog_test`, must be created to start the server successfully
* The server will return a success response if the email is not taken, passwords match, and there is no internal error.
* These endpoint requires user authentication. See #1 and #2
* Only an admin can create or update app permissions
#### What are the relevant pivotal tracker stories?
#7 
#### Screenshots (if appropriate)
#### Questions:
